packages/GisidaLite/src/index.tsx

```diff
@@ -0,0 +1,83 @@
+import { isEqual } from 'lodash';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import React from 'react';
+import ReactMapboxGl, { GeoJSONLayer, ScaleControl, ZoomControl } from 'react-mapbox-gl';
+import { FactoryParameters, Props } from 'react-mapbox-gl/lib/map';
+import { Events } from 'react-mapbox-gl/lib/map-events';
+import {
+  DEFAULT_ACCESS_TOKEN,
+  DEFAULT_ATTRIBUTION,
+  DEFAULT_CONTAINER_STYLES,
+  DEFAULT_STYLES,
+  GEOJSON_DATA
+} from './constants';
+/** ReactMapboxGlProps & mapProps props */
+const ReactMapboxGlProps = {
+  accessToken: DEFAULT_ACCESS_TOKEN,
+  attributionControl: true,
+  customAttribution: DEFAULT_ATTRIBUTION,
+  injectCSS: true
+};
+const mapProps = {
+  containerStyle: DEFAULT_CONTAINER_STYLES,
+  style: DEFAULT_STYLES
+};
+
+/** interface for  GisidaLite props */
+export interface GisidaLiteProps {
+  reactMapboxGlMapFactoryUtilConfigs: FactoryParameters;
+  mapConfigs: Props & Events;
+  mapComponents: JSX.Element[];
+  layers: JSX.Element[];
```

Comment:
Are these the right types? Isn't there a mapbox type for each of these?

Reply:
The props may be of various types, say layers can be source | layer | geojsonlayer | feature | marker | popup and many more. I felt like having JSX.Element would allow for all the layers without having todo updates every time but I think that would be a better approach

Reply:
The problem then becomes that type definitions don't help validate the inputs/props.  I would vote for creating a type for all those things e.g. `type XXX = source | layer | geojsonlayer | feature | marker | popup... etc` (or maybe mapbox type definitions already have this).  I imagine that this list does not change very often, is that the case?

Reply:
For the type issue, I think it's a bit complex than I thought. most components seem to have jsx.element as their return type and I haven't found a listing with these types. 
sample from .d.ts file
```
export declare class GeoJSONLayer extends React.Component<Props> {
    private id;
    private source;
    private layerIds;
    private buildLayerId;
    private createLayer;
    isGeoJSONSource: (source?: MapboxGL.GeoJSONSourceRaw | MapboxGL.VectorSource | MapboxGL.RasterSource | MapboxGL.RasterDemSource | MapboxGL.GeoJSONSource | MapboxGL.VideoSource | MapboxGL.ImageSource | MapboxGL.CanvasSource | undefined) => source is MapboxGL.GeoJSONSource;
    componentDidUpdate(prevProps: Props): void;
    render(): null;
}
declare const _default: <T>(props: T) => JSX.Element;
```
The same is seen for Layers, I think we may have to use JSX.Element for the moment

Reply:
Cool.  Do you think it warrants creating an issue to investigate further @kahummer ?

Reply:
Sure, adding a ticket right away

Reply:
https://github.com/onaio/js-tools/issues/175